File: projects/x-forth.c/src/lexer/span.c
#include "index.h"

struct span_t
span_union(struct span_t x, struct span_t y) {
    return (struct span_t) {
        .start = x.start.index < y.start.index ? x.start : y.start,
        .end = x.end.index > y.end.index ? x.end : y.end,
    };
}

File: packages/sexp.js/src/location/Span.ts
import { type Position } from "./Position.ts"

export type Span = {
  start: Position
  end: Position
}

export function spanUnion(lhs: Span, rhs: Span): Span {
  const start = lhs.start.index < rhs.start.index ? lhs.start : rhs.start
  const end = lhs.end.index > rhs.end.index ? lhs.end : rhs.end
  return { start, end }
}
